src/ui/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml

```diff
@@ -0,0 +1,238 @@
+        });
+
+        subscriptionInfoModel.append({
+            labelText: VPNl18n.SubscriptionManagementExpiresLabel,
+            valueText: unixToDate(VPNSubscriptionData.expiresOn),
+            type: "text",
+        });
+
+        // Subscription payment model
+        if (VPNSubscriptionData.paymentType === "credit") {
+            subscriptionPaymentModel.append({
+                labelText: VPNSubscriptionData.creditCardBrand,
+                valueText: VPNl18n.SubscriptionManagementCardLast4.arg(VPNSubscriptionData.creditCardLast4),
+                type: "creditCard",
+            });
+
+            subscriptionPaymentModel.append({
+                labelText: VPNl18n.SubscriptionManagementCardExpiresLabel,
+                valueText: VPNSubscriptionData.creditCardExpMonth + " " + VPNSubscriptionData.creditCardExpYear, // TODO: Format and localize date
+                type: "text",
+            });
+        } else {
+            subscriptionPaymentModel.append({
+                labelText: VPNl18n.SubscriptionManagementSubscriptionPlatformLabel.arg("IAP Google or Apple"), // TODO: Determine and handle IAP platforms
+                valueText: "",
+                type: "text",
+            });
+        }
+    }
+
+    function unixToDate(unixTimestamp) {
+        return new Date(unixTimestamp * 1000).toLocaleDateString(currentLanguageCode, Locale.ShortFormat);
+    }
+
+    function getPlanText(currencyCode, amount, intervalCount) {
+        const amountDisplay = (amount || 0) / 100;
+
+        const currencySymbol = getCurrencySymbolFromCode(currencyCode);
+        const localizedCurrency = Number(amountDisplay).toLocaleCurrencyString(currentLanguageCode, currencySymbol);
+        
+        if (intervalCount === 12) {
+            // {¤amount} yearly
+            return VPNl18n.SubscriptionManagementPlanValueYearly.arg(localizedCurrency);
+        } else if (intervalCount === 6) {
+            // {¤amount} half-yearly
+            return VPNl18n.SubscriptionManagementPlanValueHalfYearly.arg(localizedCurrency);
+        } else if (intervalCount === 1) {
+            // {¤amount} monthly
+            return VPNl18n.SubscriptionManagementPlanValueMonthly.arg(localizedCurrency);
+        }
+
+        // TODO: Confirm that’s really the case: If we made it here the user has a free trial.
+        return VPNl18n.FreeTrialsFreeTrialLabel;
+    }
+
+    Component.onCompleted: {
+        populateListModels();
+    }
+
+    // TODO: Get a list of currencies or find another way to localize
+    function getCurrencySymbolFromCode(code) {
+        const currencies = [
+            {
+                code: "eur",
+                sign: "€"
+            },
+            {
+                code: "usd",
+                sign: "$"
+            }
+        ];
+        const selectedCurrency = currencies.find((currency) => currency.code === code);
+
+        return selectedCurrency ? selectedCurrency.sign : code.toUpperCase();
+    }
+}
```

Comment:
I did not find an obvious way to convert the currency code that is returned from FxA to their respective currency sign. FxA relies on in [Intl.NumberFormat](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat) in their [getLocalizedCurrencyString](https://github.com/mozilla/fxa/blob/main/packages/fxa-payments-server/src/lib/formats.ts#L40-L48) method, which is not available in QML. QMLs [toLocaleCurrencyString](https://doc.qt.io/qt-5/qml-qtqml-number.html#toLocaleCurrencyString-method) method does not have the ability to convert a specific currency code to a currency sign and we do not want to change the sign based on the locale.

Is this a viable option and if that’s the case — do we have a preferred list of currencies we could use?

Edit: Just saw that `Intl.NumberFormat` has full support since Node.js `12.11.0`. So I guess I’ll move this task to Guardian.

Reply:
This will not work because the currency symbols are not unique per currency but they are strongly connected with the locales. For example, "$" for USD is OK in most the european countries, but it's "US$" in Saint Lucia where the local currency is the easter caribbean dollar (XCD, symbol  EC$).
See my PR to localize currencies.